client/components/common/SelectElement/index.vue

```diff
@@ -2,88 +2,176 @@
+          :activities="activities"
+          :selected-elements="selectedElements" />
+      </template>
+      <template v-else>
+        <v-btn
+          v-if="toggleButton"
+          @click="toggleSelectAll"
+          outlined
+          class="float-right mr-4 mb-2">
+          <v-icon class="mr-2">mdi-{{ toggleButton.icon }}</v-icon>
+          {{ toggleButton.label }}
+        </v-btn>
+        <v-progress-circular v-if="loadingContent" indeterminate class="mt-5" />
+        <content-preview
+          v-else
+          @toggle="toggleElementSelection"
+          @element:open="openElement"
+          :content-containers="contentContainers"
+          :selected="selectedElements"
+          :allowed-types="allowedTypes"
+          :multiple="multiple"
+          selectable />
+      </template>
     </template>
     <template v-slot:actions>
       <v-btn
         v-if="selectedActivity"
-        @click="selectedActivity = null"
+        @click="deselectActivity"
         text outlined
         class="mr-2">
         <v-icon>mdi-arrow-left</v-icon> Back
       </v-btn>
       <v-btn @click="close" text class="ml-1">Cancel</v-btn>
-      <v-btn @click="save" color="secondary" text>Save</v-btn>
+      <v-btn @click="save" color="secondary" text>{{ submitLabel }}</v-btn>
     </template>
   </tailor-dialog>
 </template>
 
 <script>
+import activitiesApi from '@/api/activity';
 import contentElementApi from 'client/api/contentElement';
 import ContentPreview from '@/components/common/ContentPreview';
+import flatMap from 'lodash/flatMap';
 import { getDescendants as getContainers } from '@/utils/activity';
+import { getSupportedContainers } from 'shared/activities';
 import loader from '@/components/common/loader';
+import map from 'lodash/map';
 import { mapGetters } from 'vuex';
 import SelectActivity from './SelectActivity';
+import SelectRepository from './SelectRepository';
 import sortBy from 'lodash/sortBy';
 import TailorDialog from '@/components/common/TailorDialog';
 
+const getContainerTypes = type => map(getSupportedContainers(type), 'type');
+
+const TOGGLE_BUTTON = {
+  SELECT: { label: 'Select all', icon: 'checkbox-multiple-marked-outline' },
+  DESELECT: { label: 'Deselect all', icon: 'checkbox-multiple-blank-outline' }
+};
+
 export default {
   name: 'select-element',
   props: {
     selected: { type: Array, default: () => [] },
     heading: { type: String, required: true },
+    allowedTypes: { type: Array, required: true },
     multiple: { type: Boolean, default: true },
-    allowedTypes: { type: Array, default: () => [] }
+    submitLabel: { type: String, default: 'Save' },
+    headerIcon: { type: String, default: 'mdi-toy-brick-plus-outline' },
+    useCurrentRepo: { type: Boolean, default: false }
+  },
+  data: () => {
+    return {
+      repository: null,
+      selectedActivity: null,
+      contentContainers: [],
+      selectedElements: [],
+      loadingContent: false,
+      activities: []
```

Comment:
Consider grouping selected and fetched props (i.e. repository, activities, contentContainers, selectedActivity, selectedElements)